### PR TITLE
docs: document AddressAPI

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Add method docs strs to `AddressAPI`

### How I did it

Played a "Guess the Doc Str" game with @Ninjagod1251 

### How to verify it

Read through 'em

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/api/address.py [hunk 1]
```diff
@@ -9,10 +9,23 @@
 
 @abstractdataclass
 class AddressAPI:
+    """
+    A base address API class.
```
Comment (fubuloubu) [inline src/ape/api/address.py:13]: Add something like "...all account types subclass this type (Externally-Owned Accounts and Contract Accounts)"

File: src/ape/api/address.py [hunk 2]
```diff
@@ -22,15 +35,32 @@ def provider(self) -> ProviderAPI:
...
+        """
+
         self._provider = value
 
     @property
     @abstractmethod
     def address(self) -> AddressType:
-        ...
+        """
+        The raw address type.
```
Comment (fubuloubu) [inline src/ape/api/address.py:51]: The address of this account
```suggestion
        The address of this account. Subclasses must override and provide this value.
```

File: src/ape/api/address.py [hunk 3]
```diff
@@ -22,15 +35,32 @@ def provider(self) -> ProviderAPI:
...
-        # This displays methods to IPython on `a.[TAB]` tab completion
+        """
+        Display methods to IPython on ``a.[TAB]`` tab completion.
+
+        Returns:
+            list[str]: Method names that IPython uses for tab completion.
+        """
         return [
             "address",
             "balance",
```
Comment (fubuloubu) [inline src/ape/api/address.py:66]: tbh would like to add better integration with this via the base class, but can only do that once pydantic upgrade happens

File: src/ape/api/address.py [hunk 4]
```diff
@@ -45,33 +75,86 @@ def __repr__(self) -> str:
...
+
         return self.provider.get_nonce(self.address)
 
     @property
     def balance(self) -> int:
+        """
+        The total balance of the account.
+
+        Returns:
+            `int`
```
Comment (fubuloubu) [inline src/ape/api/address.py:104]: are the ticks on purpose here? the others don't have it

File: src/ape/api/address.py [hunk 5]
```diff
@@ -45,33 +75,86 @@ def __repr__(self) -> str:
...
+        Returns:
+            bool
+        """
+
         return len(self.code) > 0
 
 
 class Address(AddressAPI):
+    """
+    A blockchain address.
```
Comment (fubuloubu) [inline src/ape/api/address.py:146]:
```suggestion
    A generic blockchain address. Typically used when we do not know the contract type at a given address, or to refer to an EOA the user doesn't personally control.
```

File: src/ape/api/address.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:51]: Should we use the word `account` here though?
The `AddressAPI` can also represent a contract, is that considered an account?

File: src/ape/api/address.py (hunk 3 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:66]: You mean the feature for IPython or the docs for it?
If the feature, we should not add docs for this.

File: src/ape/api/address.py (hunk 4 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:104]: Accident!

File: src/ape/api/address.py (hunk 5 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:146]: Ha that is insightful

File: src/ape/api/address.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:51]: Ah, well based on the comment above, I see that contracts must be commonly also referred to as an accounts

File: src/ape/api/address.py (hunk 3 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:68]: btw, I have started looking into pydantic work and it is on my list! I have rebased the branch at least.

Comment (fubuloubu) [reply src/ape/api/address.py:68]: > You mean the feature for IPython or the docs for it? If the feature, we should not add docs for this.

oh I mean the feature of integrating with IPython, but looking at methods and fields of the dataclass

File: src/ape/api/address.py [hunk 6]
```diff
@@ -9,10 +9,25 @@
 
 @abstractdataclass
 class AddressAPI:
+    """
+    A base address API class. All account-types subclass
+    this type, including externally owned accounts as well
+    as contract accounts.
```
Comment (fubuloubu) [inline src/ape/api/address.py:15]: actually there isn't any other type of account than these two lol :sweat:

**Review by fubuloubu — APPROVED** [review]

File: src/ape/api/address.py (hunk 3 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:68]: Okay cool. Well if we delete this method / approach, that is no big deal.

File: src/ape/api/address.py (hunk 6 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:15]: ha so maybe like `which includes both ...`

Comment (unparalleled-js) [reply src/ape/api/address.py:15]: I don't think this do necessarily suggests there are other potential types

Comment (unparalleled-js) [reply src/ape/api/address.py:15]: Maybe we just don't need the `including` part ?

Comment (fubuloubu) [reply src/ape/api/address.py]: yeah, sounds good

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**